public/components/presence-indicator/presence-indicators.js

```diff
@@ -29,30 +29,37 @@ function wfPresenceIndicatorsDirective($rootScope, wfPresenceService,
                                 longText: [person.firstName, person.lastName].join(" "),
                                 email: person.email
                             };
+                            const currentLocations = pr.location ? [...pr.locations, pr.location] : pr.locations
+                            const editingBodyLocations = ["body", "document"];
+                            const editingFurnitureLocations = ["furniture"];
 
-                            const location = currentState.find(p => p.clientId === pr.clientId).location;
-                            const locations = currentState.find(p => p.clientId === pr.clientId).locations || [];
-
-                            const currentLocations = location ? [...locations, location] : locations
-                            const activeEditingLocations = ["body", "document", "furniture"];
-
-                            if (activeEditingLocations.some((l) => currentLocations.includes(l)) || $scope.dontDisplayIdle) {
+                            if (editingBodyLocations.some((l) => currentLocations.includes(l)) || $scope.dontDisplayIdle) {
                                 return {
                                     ...presenceObject, ...{
                                         status: "present",
-                                        longTitle: [presenceObject.longText, "editing"].join(" - "),
-                                        shortTitle: [presenceObject.email, "editing"].join(" - "),
+                                        longTitle: [presenceObject.longText, "editing body"].join(" - "),
+                                        shortTitle: [presenceObject.email, "editing body"].join(" - "),
                                         iconPrecedence: 1
                                     }
                                 };
+                            } else if(editingFurnitureLocations.some((l) => currentLocations.includes(l))) {
+                                // the user is not editing the body, has clicked 'Save and close'
+                                return {
+                                    ...presenceObject, ...{
```

Comment:
IIUC we should be able to flatten this spread into the parent object.

Reply:
Well spotted, I was following the convention in that file and missed that.